bpf/gotracer/go_grpc.c

```diff
@@ -654,48 +716,64 @@ int beyla_uprobe_grpcFramerWriteHeaders(struct pt_regs *ctx) {
     bpf_dbg_printk(
         "framer=%llx, stream_id=%lld, framer_w_pos %llx", framer, ((u64)stream_id), framer_w_pos);
 
-    u32 stream_lookup = (u32)stream_id;
+    void *w_ptr = (void *)(framer + framer_w_pos + 16);
+    bpf_probe_read(&w_ptr, sizeof(w_ptr), (void *)(framer + framer_w_pos + 8));
 
-    grpc_client_func_invocation_t *invocation =
-        bpf_map_lookup_elem(&ongoing_streams, &stream_lookup);
+    if (!w_ptr) {
```

Comment:
This code needed to do a bit more work, since now we have to read the connection pointer as well as the stream id.